pkg/node/biz/client.go

```diff
@@ -57,26 +57,19 @@ func join(peer *signal.Peer, msg proto.JoinMsg) (interface{}, *nprotoo.Error) {
 	if err != nil {
 		log.Errorf("IslbClientOnJoin failed %v", err.Error())
 	}
-	// Send getPubs => islb
-	islb.AsyncRequest(proto.IslbGetPubs, msg.RoomInfo).Then(
-		func(result nprotoo.RawMessage) {
-			var resMsg proto.GetPubResp
-			if err := result.Unmarshal(&resMsg); err != nil {
-				log.Errorf("Unmarshal pub response %v", err)
-				return
-			}
-			log.Infof("IslbGetPubs: result=%v", resMsg)
-			for _, pub := range resMsg.Pubs {
-				if pub.MID == "" {
-					continue
-				}
-				notif := proto.StreamAddMsg(pub)
-				peer.Notify(proto.ClientOnStreamAdd, notif)
-			}
-		},
-		func(err *nprotoo.Error) {})
 
-	return emptyMap, nil
+	result, err := islb.SyncRequest(proto.IslbGetPubs, msg.RoomInfo)
```

Comment:
This will certainly help, but I'm not sure it truly solves the race condition if another client is adding a stream while this executes. This is synchronous within this client goroutine.  I say we try it because I'd rather not add another lock to getPubs in islb, but that may be necessary to solve it.